src/containers/SideNav/SideNav.js

```diff
@@ -97,70 +97,132 @@ class SideNav extends Component {
           >
             <SideNavMenuItem
               element={NavLink}
+              isActive={
+                !!matchPath(location.pathname, {
+                  path: this.getPath(urls.pipelines.all())
+                })
+              }
               to={this.getPath(urls.pipelines.all())}
             >
               Pipelines
             </SideNavMenuItem>
             <SideNavMenuItem
               element={NavLink}
+              isActive={
+                !!matchPath(location.pathname, {
+                  path: this.getPath(urls.pipelineRuns.all())
+                })
+              }
               to={this.getPath(urls.pipelineRuns.all())}
             >
               PipelineRuns
             </SideNavMenuItem>
             <SideNavMenuItem
               element={NavLink}
+              isActive={
+                !!matchPath(location.pathname, {
+                  path: this.getPath(urls.pipelineResources.all())
+                })
+              }
               to={this.getPath(urls.pipelineResources.all())}
             >
               PipelineResources
             </SideNavMenuItem>
             <SideNavMenuItem
               element={NavLink}
+              isActive={
+                !!matchPath(location.pathname, {
+                  path: this.getPath(urls.tasks.all())
+                })
+              }
               to={this.getPath(urls.tasks.all())}
             >
               Tasks
             </SideNavMenuItem>
-            <SideNavMenuItem element={NavLink} to={urls.clusterTasks.all()}>
+            <SideNavMenuItem
+              element={NavLink}
+              isActive={
+                !!matchPath(location.pathname, {
+                  path: urls.clusterTasks.all()
+                })
+              }
+              to={urls.clusterTasks.all()}
+            >
               ClusterTasks
             </SideNavMenuItem>
             <SideNavMenuItem
               element={NavLink}
+              isActive={
+                !!matchPath(location.pathname, {
+                  path: this.getPath(urls.taskRuns.all())
+                })
+              }
               to={this.getPath(urls.taskRuns.all())}
             >
               TaskRuns
             </SideNavMenuItem>
             <SideNavMenuItem
               element={NavLink}
+              isActive={
+                !!matchPath(location.pathname, {
+                  path: this.getPath(urls.conditions.all())
+                })
+              }
               to={this.getPath(urls.conditions.all())}
             >
               Conditions
             </SideNavMenuItem>
             {this.props.isTriggersInstalled && (
-              <>
```

Comment:
Due to the way the active item detection works in the Carbon `SideNavMenu`, the `SideNavMenuItem`s must be direct children of the `SideNavMenu` component and cannot be nested in a fragment or other component. This also means we cannot use a custom component that renders the `SideNavMenuItem` to reduce the duplication.